### PR TITLE
BSTOR-48: Product Refactoring and integration to cart

### PR DESCRIPTION
**Pull request by Elrhino**

### CODE REVIEW

File: client/src/main/java/com/arcbees/beeshop/client/application/product/CurrentProductImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.arcbees.beeshop.common.dto.ProductType;
+import com.arcbees.beeshop.common.dto.ProductDto;
+import com.gwtplatform.mvp.client.proxy.PlaceManager;
+import com.gwtplatform.mvp.shared.proxy.PlaceRequest;
+
+public class CurrentProductImpl implements CurrentProduct {
+    private final PlaceManager placeManager;
+
+    @Inject
+    public CurrentProductImpl(
```
Comment (olafleur) [inline client/src/main/java/com/arcbees/beeshop/client/application/product/CurrentProductImpl.java]: rm public

File: common/src/main/java/com/arcbees/beeshop/common/dto/Product.java [hunk 2]
```diff
@@ -1,92 +1,52 @@
 /*
  * Copyright 2015 ArcBees Inc.
  *
- * Licensed under the Apache License, Version 2.0 (the "License"); you may not
- * use this file except in compliance with the License. You may obtain a copy of
- * the License at
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
```
Comment (jclariviere) [inline common/src/main/java/com/arcbees/beeshop/common/dto/Product.java]: Revert license change and fix your IntelliJ :)

File: common/src/main/java/com/arcbees/beeshop/common/dto/ProductDto.java [hunk 3]
```diff
@@ -12,23 +17,6 @@ public ProductDto(Product product, Brand brand) {
         this.brand = brand;
     }
 
-    @Override
-    public boolean equals(Object o) {
```
Comment (jclariviere) [inline common/src/main/java/com/arcbees/beeshop/common/dto/ProductDto.java:16]: No longer needed?

**Comment by jclariviere** [comment]
LGTM after comments fix

**Pull request merged**